service/src/main/java/bio/terra/landingzone/stairway/flight/StepsDefinitionProvider.java

```diff
@@ -0,0 +1,28 @@
+package bio.terra.landingzone.stairway.flight;
+
+import bio.terra.landingzone.library.landingzones.definition.ArmManagers;
+import bio.terra.landingzone.library.landingzones.definition.ResourceNameGenerator;
+import bio.terra.landingzone.library.landingzones.definition.factories.ParametersResolver;
+import bio.terra.stairway.RetryRule;
+import bio.terra.stairway.Step;
+import java.util.List;
+import org.apache.commons.lang3.tuple.Pair;
+
+public interface StepsDefinitionProvider {
+  /**
+   * Returns list of required together with corresponding retry rule. Each step creates one specific
+   * landing zone resource. Order of steps matters. If a resource1 has dependency on another
+   * resource2 then resource1 should be deployed first and save its id or another required
```

Comment:
Is there a way for the steps to declare their dependencies? At the very least, it would be good to have dependencies mentioned in the JavaDoc for the individual steps.

Reply:
It is not possible now.